### ISSUE TITLE
getProp not working in Android O

### ISSUE THREAD

**Issue by Filipe-Figueiredo** [c-ares/c-ares#111]
The final APIs for Android O are out. As you may know, system properties net.dns1, net.dns2 (...) were removed from the API. So, c-Ares will not be able to get servers' addresses. An alternative will be to use ares_init_options and pass the servers as parameters, however it would be nice to have this working as previous.

This is the error in the logs:

**E/libc: Access denied finding property "net.dns1"**

Are you planning to fix this?

**Comment by daviddrysdale** [comment]
Looks like the original version of that code was added in commit b13c6552f8182736b0f05d9dc1bd0e8f3d1f4d3e in 2010.

Do you know if there's an Android API (for C, which I guess means from the NDK) for getting nameservers for O?

**Comment by Filipe-Figueiredo** [comment]
Sincerely I don't know. I will go around this by creating a jni layer to call this:   
https://developer.android.com/reference/android/net/LinkProperties.html#getDnsServers()

Then I will use ares_init_options to pass the list of servers as parameters. 
If you find a way to fix this is c-ares I will appreciate. Please keep me informed, and if I find anything I will write it here too.

**Comment by rtamin** [comment]
Hello Filipe, could you please share some more details about how you are accessing LinkProperties ?  I know using JNI layer it is possible to instantiate and call java objects from C. 

But the problem I am having is how to access the LinkProperties ? In order to access the LinkProperties I need access to the ConnectivityManager and in order to access the ConnectivityManager I need access to the Android context or Activity.

Below is a code that runs in an Android's application activity and that prints the DNS server (e.g. 127.0.0.1) using LinkProperties. 

How do I access the LinkProperties from C using JNI layer ?

```java
// Activity onCreate method
protected void onCreate(Bundle savedInstanceState) {

    ...
    ...

    ConnectivityManager cm = (ConnectivityManager) this.getSystemService("connectivity");

    for (Network network : cm.getAllNetworks()) {

        LinkProperties linkProperties = cm.getLinkProperties(network);
        List<InetAddress> dnsServers = linkProperties.getDnsServers();

        for (InetAddress inetAddress : dnsServers) {
            String ipAddress = inetAddress.getHostAddress();
            Log.d(TAG, String.format("DNS Server = [%s]", ipAddress));
        }
    }
}
```

**Comment by Filipe-Figueiredo** [comment]
Hello,

You don't need an activity to access LinkProperties. You just need to have the application Context saved somewhere:

`public List<String> getDNSServersList() {
		List<String> result = new ArrayList<>();

		if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.M)
			return null;

		ConnectivityManager connectivityManager = (ConnectivityManager) MyApplication.getContext().getSystemService(Context.CONNECTIVITY_SERVICE);

		Network network = connectivityManager.getActiveNetwork();

		LinkProperties linkProperties = connectivityManager.getLinkProperties(network);
		
                for (InetAddress address : linkProperties.getDnsServers())
	               result.add(address.getHostAddress());

		return result;
	}`

Hope this helps.

**Comment by user-none** [comment]
Patch in pull #148

**Comment by bradh352** [comment]
merged #148
